### PR TITLE
Enhance contract deploy constructor

### PR DESCRIPTION
**Pull request by ququzone**
**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

#195

### CODE REVIEW

File: src/contract/contract.ts [hunk 1]
```diff
@@ -111,6 +117,9 @@ export class Contract {
 
   public async deploy(
     account: IAccount,
+    // @ts-ignore
+    // tslint:disable-next-line: no-any
+    inputs: Array[any],
```
Comment (puncsky) [inline src/contract/contract.ts]: Array<any> or Array<Array<any>>?

File: src/contract/contract.ts [hunk 2]
```diff
@@ -121,12 +130,34 @@ export class Contract {
       throw new Error("no rpc method provider specified");
     }
 
+    let data = this.options.data || Buffer.from([]);
+    if (this.abi && this.abi.hasOwnProperty("constructor")) {
+      /* tslint:disable:no-string-literal */
+      const abiFunc = this.abi["constructor"];
+      const userInput = {};
```
Comment (puncsky) [inline src/contract/contract.ts]: add type def

File: src/contract/__test__/contract-interact.ts [hunk 3]
```diff
@@ -111,6 +111,7 @@ test.skip("Contract_method_get_SimpleStorage", async t => {
 
   const client = new RpcMethod(TEST_HOSTNAME, { timeout: 10000 });
   const contract = new Contract(
+    // @ts-ignore
```
Comment (puncsky) [inline src/contract/__test__/contract-interact.ts]: add missing param

File: src/contract/__test__/contract-interact.ts [hunk 4]
```diff
@@ -135,6 +136,7 @@ test.skip("Contract_method_set_SimpleStorage", async t => {
   const client = new RpcMethod(TEST_HOSTNAME, { timeout: 10000 });
 
   const contract = new Contract(
+    // @ts-ignore
```
Comment (puncsky) [inline src/contract/__test__/contract-interact.ts]: add missing param

File: src/contract/contract.ts (hunk 1 above)
Comment (ququzone) [reply src/contract/contract.ts]: Array params

**Pull request merged**